samples/filter-features-in-scene/src/main/java/com/esri/arcgismaps/sample/filterfeaturesinscene/components/FilterFeaturesInSceneViewModel.kt

```diff
@@ -0,0 +1,133 @@
+    val arcGISScene: ArcGISScene by mutableStateOf(ArcGISScene(BasemapStyle.ArcGISTopographic).apply {
+        // Add an elevation source to the scene's base surface.
+        baseSurface.elevationSources.add(ArcGISTiledElevationSource(uri = "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"))
+        // OSM building layer
+        operationalLayers.add(osmBuildingsSceneLayer)
+        // Add the buildings scene layer to the operational layers
+        operationalLayers.add(sanFranciscoBuildingsSceneLayer)
+        // Set the initial viewpoint of the scene
+        initialViewpoint = Viewpoint(
+            Point(0.0, 0.0), Camera(
+                latitude = 37.7041, longitude = -122.421, altitude = 207.0, heading = 60.0, pitch = 70.0, roll = 0.0
+            )
+        )
+    })
+
+    // Create a boundary polygon that will be populated with the extent of the San Francisco buildings layer once loaded
+    private val sanFranciscoBuildingsBoundary = runBlocking { createBoundaryPolygon() }
+
+    // Set up graphic overlay and graphic for the San Francisco buildings layer extent. It is assigned to the SceneView
+    // in the composable function
+    val graphicsOverlay = GraphicsOverlay().apply {
+        graphics.add(
+            Graphic(
+                sanFranciscoBuildingsBoundary, SimpleFillSymbol(
+                    style = SimpleFillSymbolStyle.Solid, color = Color.transparent, outline = SimpleLineSymbol(
+                        style = SimpleLineSymbolStyle.Solid, color = Color.red, width = 5.0f
+                    )
+                )
+            )
+        )
+    }
+
+    /**
+     * One the San Francisco buildings layer is loaded, create and return a boundary [Polygon].
+     */
+    private suspend fun createBoundaryPolygon(): Polygon? {
+        // Load the San Francisco buildings layer
+        sanFranciscoBuildingsSceneLayer.load().onSuccess {
+            // Create a polygon boundary using the San Francisco buildings layer extent
+            return sanFranciscoBuildingsSceneLayer.fullExtent?.let {
+                PolygonBuilder().apply {
+                    addPoint(it.xMin, it.yMin)
+                    addPoint(it.xMax, it.yMin)
+                    addPoint(it.xMax, it.yMax)
+                    addPoint(it.xMin, it.yMax)
+                }.toGeometry()
+            }
+        }.onFailure {
+            messageDialogVM.showMessageDialog(it.message.toString(), it.cause.toString())
+        }
+        // Return null if the layer doesn't load or the extent is not available
+        return null
+    }
+
+    /**
+     * Filter the OSM buildings layer to only show buildings that are disjoint from the San Francisco buildings layer
+     * extent.
+     */
+    fun filterScene() {
+        // Check that the San Francisco buildings layer extent is available
+        sanFranciscoBuildingsBoundary?.let { boundary ->
```

Comment:
With the above changes this code could now use: 
```suggestion
        boundaryGraphic.geometry?.let { boundary ->
```